src/aiodynamo/client.py

```diff
@@ -583,38 +625,79 @@ def table(self, name: TableName) -> Table:
         update_expression: UpdateExpression,
         *,
         return_values: ReturnValues = ReturnValues.none,
-        condition: ConditionBase = None
+        condition: Condition = None,
     ) -> Union[Item, None]:
+        params = Parameters()
 
-        (
-            update_expression,
-            expression_attribute_names,
-            expression_attribute_values,
-        ) = update_expression.encode()
-
-        if not update_expression:
+        expression = update_expression.encode(params)
+        if not expression:
             raise EmptyItem()
 
-        builder = ConditionExpressionBuilder()
+        payload = {
+            "TableName": table,
+            "Key": py2dy(key),
+            "UpdateExpression": expression,
+            "ReturnValues": return_values.value,
+        }
         if condition:
-            condition_expression, ean, eav = builder.build_expression(condition)
-            expression_attribute_names.update(ean)
-            expression_attribute_values.update(eav)
-        else:
-            condition_expression = None
-        resp = await self.core.update_item(
-            **clean(
-                TableName=table,
-                Key=py2dy(key),
-                UpdateExpression=update_expression,
-                ExpressionAttributeNames=expression_attribute_names,
-                ExpressionAttributeValues=py2dy(expression_attribute_values),
-                ConditionExpression=condition_expression,
-                ReturnValues=return_values.value,
-            )
-        )
+            payload["ConditionExpression"] = condition.encode(params)
+
+        payload["ExpressionAttributeNames"] = params.get_expression_names()
+        payload["ExpressionAttributeValues"] = params.get_expression_values()
```

Comment:
likewise, is it possible to update expression to be essentially empty, and in that case, is it correct to add attribute names/values fields?

Reply:
if Update Expression is empty, it raises an exception in line 636.